src/StringViews.jl

```diff
@@ -58,6 +58,7 @@ Base.unsafe_convert(::Type{Ptr{UInt8}}, s::DenseStringViewAndSub) = pointer(s)
 Base.unsafe_convert(::Type{Ptr{Int8}}, s::DenseStringViewAndSub) = convert(Ptr{Int8}, pointer(s))
 Base.cconvert(::Type{Ptr{UInt8}}, s::DenseStringViewAndSub) = s
 Base.cconvert(::Type{Ptr{Int8}}, s::DenseStringViewAndSub) = s
+Base.convert(::Type{T}, s::StringView{T}) where {T<:AbstractVector{UInt8}} = s.data
```

Comment:
This seems wrong to me.   It allows you to implicitly convert a `StringView` to an array of `UInt8`, which we don't do for any other string type.

Reply:
It's true that Strings don't allow for implicit conversion to `Vector{UInt8}`, which is good. They have no natural relationship to the type, and it would be allocating.  Nor is `convert(Base.CodeUnits{UInt8, String}, "abc")` defined, which I find a bit surprising, but there's little to no need to store that construct, so I don't find the lack dispositive.

The `convert` I've added is non-allocating, and does have a natural relationship to the type: it's the only data in the struct, and the conversion type is the parameter of the `StringView`. It only allows conversion into the type of the underlying data, so it works like this:

```julia
julia> sv = StringView("abcd")
"abcd"

julia> convert(Vector{UInt8}, sv)
MethodError: Cannot `convert` an object of type StringView{Base.CodeUnits{UInt8, String}} to an object of type Vector{UInt8}

julia> convert(Base.CodeUnits{UInt8, String}, sv)
4-element Base.CodeUnits{UInt8, String}:
 0x61
 0x62
 0x63
 0x64
```
I think this is reasonable behavior for a wrapper type. 

I added it so that [this implementation](https://github.com/JuliaLang/julia/blob/f4866b7617a3d214e6e4437e13fd8b389c64b078/base/number.jl#L380C1-L381C41) of `oneunit` would function correctly:

```
oneunit(x::T) where {T} = T(one(x))
oneunit(::Type{T}) where {T} = T(one(T))
```

And I was ok with that decision, because converting a wrapper to its underlying data seems like a reasonable operation to have.

However, if you aren't persuaded, that's fine, it also works to add the following:

```julia
StringView{S}(s::StringView{S}) where {S<:AbstractVector{UInt8}} = s
```
With that added, and the `convert` commented out, all the same tests pass. 

I'll make a case for including both, but it's a weakly-held opinion. I've gotten good mileage out of adding unwrapping `convert` to data types which consist soley of another type, but it might not be the right choice here.

Reply:
I would prefer just the `StringView{S}(s::StringView{S})` constructor.